framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1958,8 +1959,13 @@ private static void putOrAddToMap(
     }
 
     /**
-     * Just like Map.put, but merges with any existing annotated type for the given key, instead of
-     * replacing it.
+     * Just like Map.put, but merges (using {@link AnnotatedTypeMerger#merge}) with any existing
+     * annotated type for the given key, instead of replacing it.
+     *
+     * @param m a map
+     * @param key a key for the map
+     * @param newType a value to merge into the map: merge it with the value currently at {@code
+     *     key} and replace that value
      */
     private void putNew(
```

Comment:
I realize this is not your fault, but the name `putNew` isn't descriptive and even sounds like the opposite of what this does.  How about `putMerge` or some better name?